pkg/webhook/service_account.go

```diff
@@ -40,12 +41,16 @@ func CreateWebhookSA(client crclient.Client,
 		if err != nil {
 			return err
 		}
-		serviceAccount.ResourceVersion = existingCfg.ResourceVersion
-		err = client.Update(ctx, serviceAccount)
-		if err != nil {
-			return err
+		if needsUpdate(serviceAccount, existingCfg) {
+			serviceAccount.ResourceVersion = existingCfg.ResourceVersion
+			err = client.Update(ctx, serviceAccount)
```

Comment:
Does it mean that after we update labels on SA, new token will be created? Can we avoid it with patching or it's by design for k8s?

Reply:
The `.secrets` field in a SA object gets populated with a reference to the secret that should be used. When we apply the spec SA as an update, it wipes out that field and k8s fills it again... with a new secret. 

We could potentially work around this by copying the current secrets field over to the spec before updating, but I'm not sure if that causes any issues in other ways, so I opted to avoid updating unless necessary (which realistically, should not be needed 99% of the time)